src/Likvido.ApiAuth.Common/UserInfoHeaderSerializer.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Text.Json;
+
+namespace Likvido.ApiAuth.Common
+{
+    public static class UserInfoHeaderSerializer
+    {
+        private static readonly JsonSerializerOptions SerializerOptions = new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            MaxDepth = 3,
+            WriteIndented = false
+        };
+
+        private static readonly JsonDocumentOptions DocumentOptions = new JsonDocumentOptions
+        {
+            AllowTrailingCommas = true,
+            CommentHandling = JsonCommentHandling.Skip,
+            MaxDepth = 3
+        };
+
+        public static ClaimsIdentity Deserialize(string value, TokenClaims tokenClaims = null, string authType = AuthConstants.UserInfoHeader)
+        {
+            tokenClaims = tokenClaims ?? TokenClaims.Default;
+            var rawString = Encoding.UTF8.GetString(Convert.FromBase64String(value));
+            var doc = JsonDocument.Parse(rawString, DocumentOptions);
+            var claims = new List<Claim>();
+            var jsonUserId = doc.RootElement.GetProperty("userId").GetString();
```

Comment:
Maybe we should handle the case where no `userId` has been added? - throw a specific exception?

Reply:
right made deserialization stricter with better errors